static/js/components/CourseList.js

```diff
@@ -1,22 +1,31 @@
 import React from 'react';
 import _ from 'lodash';
+import Button from 'react-bootstrap/lib/Button';
 
 import {
   makeCourseStatusDisplay,
   makeCourseProgressDisplay,
 } from '../util/util';
+import { STATUS_PASSED } from '../constants';
 
 class CourseList extends React.Component {
   render() {
     const { dashboard } = this.props;
+    let coursesPassed = 0;
+    let totalCourses = 0;
 
     let programs = dashboard.programs.map(program => {
       let sortedCourses = _.sortBy(program.courses, 'position_in_program');
+      totalCourses = program.courses.length > 0 ? program.courses.length : 0;
```

Comment:
Sorry i misunderstood edge case, where a program has no course. Now just saw api it was returning empty course list. Now has been fixed.